src/nft/ZoraCreator1155Impl.sol

```diff
@@ -386,6 +420,37 @@ contract ZoraCreator1155Impl is
         emit Purchased(msg.sender, address(minter), tokenId, quantity, msg.value);
     }
 
+    /// @notice Get the creator reward recipient address
+    /// @dev The creator is not enforced to set a funds recipient address, so in that case the reward would be claimable by creator's contract
+    function getCreatorRewardRecipient() public view returns (address payable) {
+        return config.fundsRecipient != address(0) ? config.fundsRecipient : payable(address(this));
+    }
+
+    /// @notice Mint tokens and payout rewards given a minter contract, minter arguments, a finder, and a origin
+    /// @param minter The minter contract to use
+    /// @param tokenId The token ID to mint
+    /// @param quantity The quantity of tokens to mint
+    /// @param minterArguments The arguments to pass to the minter
+    /// @param mintReferral The referrer of the mint
+    function mintWithRewards(
+        IMinter1155 minter,
+        uint256 tokenId,
+        uint256 quantity,
+        bytes calldata minterArguments,
+        address mintReferral
```

Comment:
how about just including this in minterArguments? then we wouldn't need the function overload and could save on contract size; in the `mint` function you could check for the presence of `mintReferral` in the `minterArguments`